src/subscriber.ts

```diff
@@ -447,24 +448,50 @@ export class Subscriber extends EventEmitter {
   private _constructSpan(message: Message): Span | undefined {
     // Handle cases where OpenTelemetry is disabled or no span context was sent through message
     if (
-      !this._tracing ||
+      !this._useOpentelemetry ||
       !message.attributes ||
       !message.attributes['googclient_OpenTelemetrySpanContext']
     ) {
       return undefined;
     }
+
     const spanValue = message.attributes['googclient_OpenTelemetrySpanContext'];
     const parentSpanContext: SpanContext | undefined = spanValue
       ? JSON.parse(spanValue)
       : undefined;
     const spanAttributes = {
       ackId: message.ackId,
       deliveryAttempt: message.deliveryAttempt,
+      //
+      // based on https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#topic-with-multiple-consumers
+      [MessagingAttribute.MESSAGING_SYSTEM]: 'pubsub',
+      [MessagingAttribute.MESSAGING_OPERATION]: 'process',
+      [MessagingAttribute.MESSAGING_DESTINATION]: this.name,
+      [MessagingAttribute.MESSAGING_DESTINATION_KIND]: 'topic',
+      [MessagingAttribute.MESSAGING_MESSAGE_ID]: message.id,
+      [MessagingAttribute.MESSAGING_PROTOCOL]: 'pubsub',
+      [MessagingAttribute.MESSAGING_MESSAGE_PAYLOAD_COMPRESSED_SIZE_BYTES]: 0,
+      [MessagingAttribute.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES]: message.length,
+      // Not in Opentelemetry semantic convention but mimics naming
+      'messaging.consumer.published_at':
+        typeof message.publishTime.getTime !== 'undefined'
+          ? message.publishTime.getTime()
+          : undefined,
+      'messaging.consumer.received_at': message.received,
+      'messaging.consumer.acknowlege_id': message.ackId,
+      'messaging.consumer.delivery_attempt': message.deliveryAttempt,
     };
+
     // Subscriber spans should always have a publisher span as a parent.
     // Return undefined if no parent is provided
+    const spanName = `${this.name} receive`;
```

Comment:
The operation name here must match `messaging.operation` attribute https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/messaging.md#user-content-span-name:~:text=The%20values%20allowed%20for%20%3Coperation%20name%3E,defined%20for%20message%20consumer%20spans%20below.

Reply:
Updated